src/Controls/tests/BindingSourceGen.UnitTests/BindingRepresentationGenTests.cs

```diff
@@ -72,6 +56,264 @@ public void GenerateBindingWithNestedProperties()
+    {
+        var source = """
+        using Microsoft.Maui.Controls;
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (Foo f) => f.Value);
+
+        class Foo
+        {
+            public int? Value { get; set; }
+        }
+        """;
+
+        var actualBinding = SourceGenHelpers.GetBinding(source) with { Id = 0 }; // TODO: Improve indexing of bindings
+        var expectedBinding = new CodeWriterBinding(
+                0,
+                new SourceCodeLocation("", 3, 7),
+                new TypeName("global::Foo", false, false),
+                new TypeName("int", true, false),
+                [
+                    new PathPart("f", false),
+                    new PathPart("Value", true),
+                ],
+                true
+            );
+
+        //TODO: Change arrays to custom collections implementing IEquatable
+        Assert.Equal(expectedBinding.Path, actualBinding.Path);
+        Assert.Equivalent(expectedBinding, actualBinding, strict: true);
+    }
+
+    [Fact]
+    public void GenerateBindingWithNullableSourceReferenceAndNullableReferenceElementInPathWhenNullableEnabled()
+    {
+        var source = """
+        using Microsoft.Maui.Controls;
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (Button? b) => b?.Text?.Length);
+        """;
+
+        var actualBinding = SourceGenHelpers.GetBinding(source) with { Id = 0 }; // TODO: Improve indexing of bindings
+        var expectedBinding = new CodeWriterBinding(
+                0,
+                new SourceCodeLocation("", 3, 7),
+                new TypeName("global::Microsoft.Maui.Controls.Button", true, false),
+                new TypeName("int", true, false),
+                [
+                    new PathPart("b", true),
+                    new PathPart("Text", true),
+                    new PathPart("Length", false),
+                ],
+                true
+            );
+
+        //TODO: Change arrays to custom collections implementing IEquatable
+        Assert.Equal(expectedBinding.Path, actualBinding.Path);
+        Assert.Equivalent(expectedBinding, actualBinding, strict: true);
+    }
+
+    [Fact]
+    public void GenerateBindingWithNullableReferenceTypesWhenNullableDisabled()
+    {
+        var source = """
+        using Microsoft.Maui.Controls;
+        #nullable disable
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (Foo f) => f.Bar.Length);
+
+        class Foo
+        {
+            public string Bar { get; set; }
+        }
+        """;
+
+        var actualBinding = SourceGenHelpers.GetBinding(source) with { Id = 0 };
+        var expectedBinding = new CodeWriterBinding(
+                0,
+                new SourceCodeLocation("", 4, 7),
+                new TypeName("global::Foo", true, false),
+                new TypeName("int", false, false), // Should it be false?
```

Comment:
When nullable is disabled and we have a reference type (which is then nullable), its value types properties are not null right?

Reply:
yes, it would have to be `Nullable<int>`